> > @tuananh I was following `CONTRIBUTING.md` but somehow I couldn't able to get success build from _any_ package due to `ERROR: unable to select packages:` error. I'm on Mac. Am I missing something? Should I use lima instead `dev-container`?
> 
> can you post more error logs?

Where is full logs:

<details>

  ```
  [sdk] ❯ make packages/kubevela
Makefile:166: warning: overriding recipe for target 'packages/x86_64/http-parser-2.9.4-r0.apk'
Makefile:164: warning: ignoring old recipe for target 'packages/x86_64/http-parser-2.9.4-r0.apk'
Makefile:243: warning: overriding recipe for target 'packages/x86_64/wasi-libc-0.20220525-r0.apk'
Makefile:169: warning: ignoring old recipe for target 'packages/x86_64/wasi-libc-0.20220525-r0.apk'
mkdir -p ./kubevela/
SOURCE_DATE_EPOCH=0 /usr/bin/melange build kubevela.yaml --repository-append /Users/furkan.turkal/src/public/os/packages --keyring-append local-melange.rsa.pub --signing-key local-melange.rsa --pipeline-dir /usr/share/melange/pipelines --arch x86_64 --env-file build-x86_64.env --cache-dir gs://wolfi-sources/ --namespace wolfi  --source-dir ./kubevela/
2023/01/13 09:20:14 building for [amd64]
2023/01/13 09:20:14 melange: detected git commit for build configuration: 68d6405081f4f7dcb7fd5491ed5d6cae3c4312af
2023/01/13 09:20:14 melange (kubevela/x86_64): melange is building:
2023/01/13 09:20:14 melange (kubevela/x86_64):   configuration file: kubevela.yaml
2023/01/13 09:20:14 melange (kubevela/x86_64):   workspace dir: /tmp/melange-workspace-3823748281
2023/01/13 09:20:14 melange (kubevela/x86_64): evaluating pipelines for package requirements
2023/01/13 09:20:14 melange (kubevela/x86_64):   adding package "wget" for pipeline "Fetch and extract external object into workspace"
2023/01/13 09:20:14 melange (kubevela/x86_64): building workspace in '/tmp/melange-guest-1246189302' with apko
Jan 13 09:20:14.096 [INFO] [arch:x86_64] build context:
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   working directory: /tmp/melange-guest-1246189302
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   tarball path:
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   use proot: false
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   source date: 1970-01-01 00:00:00 +0000 UTC
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   Docker mediatypes: false
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   SBOM output path:
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   arch: x86_64
Jan 13 09:20:14.096 [INFO] [arch:x86_64] image configuration:
Jan 13 09:20:14.096 [INFO] [arch:x86_64]   contents:
Jan 13 09:20:14.096 [INFO] [arch:x86_64]     repositories: []
Jan 13 09:20:14.096 [INFO] [arch:x86_64]     keyring:      []
Jan 13 09:20:14.097 [INFO] [arch:x86_64]     packages:     [build-base busybox ca-certificates-bundle git go wget]
Jan 13 09:20:14.097 [INFO] [arch:x86_64]   accounts:
Jan 13 09:20:14.097 [INFO] [arch:x86_64]     runas:
Jan 13 09:20:14.097 [INFO] [arch:x86_64]     users:
Jan 13 09:20:14.097 [INFO] [arch:x86_64]       - uid=1000(build) gid=1000
Jan 13 09:20:14.097 [INFO] [arch:x86_64]     groups:
Jan 13 09:20:14.097 [INFO] [arch:x86_64]       - gid=1000(build) members=[build]
Jan 13 09:20:14.097 [INFO] [arch:x86_64] doing pre-flight checks
Jan 13 09:20:14.097 [INFO] [arch:x86_64] building image fileystem in /tmp/melange-guest-1246189302
Jan 13 09:20:14.097 [INFO] [arch:x86_64] initializing apk database
Jan 13 09:20:14.097 [INFO] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] running: /sbin/apk add --initdb --arch x86_64 --root /tmp/melange-guest-1246189302
Jan 13 09:20:14.136 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] OK: 0 MiB in 0 packages
Jan 13 09:20:14.140 [INFO] [arch:x86_64] initializing apk keyring
Jan 13 09:20:14.141 [INFO] [arch:x86_64] initializing apk repositories
Jan 13 09:20:14.140 [INFO] [arch:x86_64] initializing apk world
Jan 13 09:20:14.150 [DEBUG] [arch:x86_64] appending 1 extra keys to keyring
Jan 13 09:20:14.150 [DEBUG] [arch:x86_64] installing key local-melange.rsa.pub
Jan 13 09:20:14.150 [DEBUG] [arch:x86_64] installing key /usr/share/apk/keys/x86_64/alpine-devel@lists.alpinelinux.org-4a6a0840.rsa.pub
Jan 13 09:20:14.150 [DEBUG] [arch:x86_64] installing key /usr/share/apk/keys/x86_64/alpine-devel@lists.alpinelinux.org-6165ee59.rsa.pub
Jan 13 09:20:14.150 [DEBUG] [arch:x86_64] installing key /usr/share/apk/keys/x86_64/alpine-devel@lists.alpinelinux.org-5261cecb.rsa.pub
Jan 13 09:20:14.152 [INFO] [arch:x86_64] synchronizing with desired apk world
Jan 13 09:20:14.152 [INFO] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] running: /sbin/apk fix --root /tmp/melange-guest-1246189302 --no-scripts --no-cache --update-cache --arch x86_64
Jan 13 09:20:14.158 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] WARNING: Ignoring /Users/furkan.turkal/src/public/os/packages: No such file or directory
Jan 13 09:20:14.158 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:] ERROR: unable to select packages:
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   build-base (no such package):
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[build-base]
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   busybox (no such package):
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[busybox]
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   ca-certificates-bundle (no such package):
Jan 13 09:20:14.159 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[ca-certificates-bundle]
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   git (no such package):
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[git]
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   go (no such package):
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[go]
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]   wget (no such package):
Jan 13 09:20:14.160 [DEBUG] [arch:x86_64] [cmd:apk] [use-proot:false] [use-qemu:]     required by: world[wget]
2023/01/13 09:20:14 ERROR: failed to build package. the build environment has been preserved:
2023/01/13 09:20:14 melange (kubevela/x86_64):   workspace dir: /tmp/melange-workspace-3823748281
2023/01/13 09:20:14 melange (kubevela/x86_64):   guest dir: /tmp/melange-guest-1246189302
Error: failed to build package: unable to build guest: unable to generate image: initializing apk: failed to fixate apk world: exit status 6
2023/01/13 09:20:14 error during command execution: failed to build package: unable to build guest: unable to generate image: initializing apk: failed to fixate apk world: exit status 6
make: *** [Makefile:353: packages/x86_64/kubevela-1.6.7-r0.apk] Error 1
  ```
</details>

--- a/Makefile
+++ b/Makefile
@@ -368,6 +368,7 @@ $(eval $(call build-package,py3-urllib3,1.26.13-r0))
 $(eval $(call build-package,py3-yaml,6.0-r0))
 $(eval $(call build-package,aws-cli,1.27.38-r0))
 $(eval $(call build-package,kubescape,2.0.183-r0))
+$(eval $(call build-package,kubevela,1.7.0-r0))
 
 .build-packages: ${PACKAGES}
 

--- a/kubevela.yaml
+++ b/kubevela.yaml
@@ -1,0 +1,45 @@
+package:
+  name: kubevela
+  version: 1.7.0
+  epoch: 0
+  description: KubeVela is a modern application delivery platform that makes deploying and operating applications across today's hybrid, multi-cloud environments easier, faster and more reliable
+  target-architecture:
+    - all
+  copyright:
+    - license: Apache-2.0
+      paths:
+        - "*"
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - go
+      - build-base
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/kubevela/kubevela/archive/refs/tags/v${{package.version}}.tar.gz
+      expected-sha256: fd102cc6fd76273d5a3bc88faa050f3d0eed5cbfc9a253522d9fad012cc0babf
+  - runs: |
+      export CGO_ENABLED=0
+      export GO111MODULE=on
+  - uses: go/build
+    with:
+      packages: ./cmd/core/main.go
+      ldflags: '"-X github.com/oam-dev/kubevela/version.VelaVersion=${{package.version}}"'
+      output: manager
+  - uses: go/build
+    with:
+      packages: ./cmd/apiserver/main.go
+      ldflags: '"-X github.com/oam-dev/kubevela/version.VelaVersion=${{package.version}}"'
+      output: apiserver
+  - uses: go/build
+    with:
+      packages: ./references/cmd/cli/main.go 
+      ldflags: '"-X github.com/oam-dev/kubevela/version.VelaVersion=${{package.version}}"'
+      output: vela
+  - uses: strip

